src/pyuvdata/uvbeam/analytic_beam.py

```diff
@@ -0,0 +1,899 @@
+                            )
+                        a_equal = False
+                    elif np.issubdtype(self_attr.dtype, np.number) and not np.allclose(
+                        self_attr, other_attr
+                    ):
+                        if not silent:
+                            print(
+                                f"attribute {attr} does not match using np.allclose. "
+                                f"Left is {self_attr}, right is {other_attr}."
+                            )
+                        a_equal = False
+                    elif self_attr.tolist() != other_attr.tolist():
+                        if not silent:
+                            print(
+                                f"attribute {attr} does not match after "
+                                "converting string numpy array to list. "
+                                f"Left is {self_attr}, right is {other_attr}."
+                            )
+                        a_equal = False
+                elif self_attr != other_attr:
+                    if not silent:
+                        print(
+                            f"attribute {attr} does not match. Left is "
+                            f"{self_attr}, right is {other_attr}."
+                        )
+                    a_equal = False
+
+            return a_equal
+        else:
+            if not silent:
+                print("Classes do not match")
+            return False
+
+    def __ne__(self, other, *, check_extra=True, silent=True):
+        """
+        Test if classes match and parameters are not equal.
+
+        Parameters
+        ----------
+        other : class
+            Other class instance to check
+        silent : bool
+            Option to turn off printing explanations of why equality fails. Useful to
+            prevent __ne__ from printing lots of messages.
+
+        Returns
+        -------
+        bool
+            True if the two instances are equivalent.
+
+        """
+        return not self.__eq__(other, silent=silent)
+
+    def _check_eval_inputs(
+        self,
+        *,
+        az_array: npt.NDArray[np.float],
+        za_array: npt.NDArray[np.float],
+        freq_array: npt.NDArray[np.float],
+    ):
+        """Check the inputs for the eval methods."""
+        if az_array.ndim > 1 or za_array.ndim > 1 or freq_array.ndim > 1:
+            raise ValueError(
+                "az_array, za_array and freq_array must all be one dimensional."
+            )
+
+        if az_array.shape != za_array.shape:
+            raise ValueError("az_array and za_array must have the same shape.")
+
+    def _get_empty_data_array(
+        self, npts: int, nfreqs: int, beam_type: str = "efield"
+    ) -> npt.NDArray[np.float]:
+        """Get the empty data to fill in the eval methods."""
+        if beam_type == "efield":
+            return np.zeros((self.Naxes_vec, self.Nfeeds, nfreqs, npts), dtype=float)
```

Comment:
hmm. I think it can in principle be complex for efield beams or for cross-pol power beams, so I will change it. Currently all our analytic beams are real, so I didn't catch it, thanks for finding it.